Mergin/utils.py

```diff
@@ -700,63 +723,99 @@ def save_raster_layer(raster_layer, project_dir):
     elif driver_name == "GPKG":
         save_raster_to_geopackage(raster_layer, project_dir)
     else:
         save_raster_as_geotif(raster_layer, project_dir)
 
 
-def copy_tif_raster(raster_layer, project_dir):
-    """Create a copy of raster layer file(s) without rewriting."""
-    dp = raster_layer.dataProvider()
-    src_filepath = dp.dataSourceUri() if os.path.isfile(dp.dataSourceUri()) else None
-    if src_filepath is None:
-        raise PackagingError(f"Can't find the source file for {raster_layer.name()}")
-
-    # Make sure the destination path is unique and create a copy including any external pyramids
-    new_raster_filename = get_unique_filename(os.path.join(project_dir, os.path.basename(src_filepath)))
-    shutil.copy(src_filepath, new_raster_filename)
-    # External pyramids
-    if os.path.exists(src_filepath + ".ovr"):
-        shutil.copy(src_filepath + ".ovr", new_raster_filename + ".ovr")
-    # Erdas external pyramids
-    src_filepath_no_ext = os.path.splitext(src_filepath)[0]
-    if os.path.exists(src_filepath_no_ext + ".aux"):
-        shutil.copy(src_filepath_no_ext + ".aux", os.path.splitext(new_raster_filename)[0] + ".aux")
+def copy_layer_files(layer, src_path, project_dir):
+    """
+    Creates a copy of the layer file(s) in the MerginMaps project directory
+    and updates layer datasource to point to the correct location.
+
+    If necessary, auxilary files (e.g. world files, overviews, etc) are also
+    copied to the MerginMaps project directory.
+    """
+    if not os.path.exists(src_path):
+        raise PackagingError(f"Can't find the source file for {layer.name()}")
+
+    # Make sure the destination path is unique by adding suffix to it if necessary
+    new_filename = get_unique_filename(os.path.join(project_dir, os.path.basename(src_path)))
+    shutil.copy(src_path, new_filename)
+
+    # for GDAL rasters copy overviews and any other auxilary files
+    if layer.dataProvider().name() == "gdal":
+        copy_gdal_aux_files(src_path, new_filename)
+
+    # Update layer datasource so the layer is loaded from the new location
+    update_datasource(layer, new_filename)
+
+
+def update_datasource(layer, new_path):
+    """Updates layer datasource, so the layer is loaded from the new location"""
+    options = QgsDataProvider.ProviderOptions()
+    options.layerName = layer.name()
+    if layer.dataProvider().name() == "vectortile":
+        layer.setDataSource(f"url={new_path}&type=mbtiles", layer.name(), layer.dataProvider().name(), options)
+    elif layer.dataProvider().name() == "wms":
+        layer.setDataSource(f"url=file://{new_path}&type=mbtiles", layer.name(), layer.dataProvider().name(), options)
+    else:
+        layer.setDataSource(new_path, layer.name(), layer.dataProvider().name(), options)
+
+
+def copy_gdal_aux_files(src_path, new_path):
+    """
+    Copies various auxilary files created/used by GDAL, e.g. pyramids,
```

Comment:
is this the full list?

also this can be refactored as logic seems quite similar